DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -193,77 +201,75 @@ List<DelegateUser> delegateUsers
-                    myAccountDetailsData.Surname,
-                    myAccountDetailsData.Email,
-                    myAccountDetailsData.ProfileImage,
-                    verifiedAdminUser.Id
-                );
-            }
+            userDataService.SetCentreEmail(
+                userId,
+                centreAnswersData.CentreId,
+                centreAnswersData.CentreEmail
+            );
 
-            if (verifiedDelegateUsers.Count != 0)
+            if (delegateUsers.Count != 0)
             {
-                userDataService.UpdateUser(
-                    myAccountDetailsData.FirstName,
-                    myAccountDetailsData.Surname,
-                    myAccountDetailsData.Email,
-                    myAccountDetailsData.ProfileImage,
-                    myAccountDetailsData.ProfessionalRegistrationNumber,
-                    myAccountDetailsData.HasBeenPromptedForPrn,
-                    centreAnswersData.JobGroupId,
-                    1 // TODO HEEDLS-887 This needs correcting to the correct UserId for the delegate record.
+                var delegateId = myAccountDetailsData.DelegateId!.Value;
+                // TODO HEEDLS-887 Remove this call once groupsService.SynchroniseUserChangesWithGroups is refactored
+                var oldDelegateUserDetails = userDataService.GetDelegateUserById(delegateId);
+
+                userDataService.UpdateDelegateUserCentrePrompts(
+                    delegateId,
+                    centreAnswersData.Answer1,
+                    centreAnswersData.Answer2,
+                    centreAnswersData.Answer3,
+                    centreAnswersData.Answer4,
+                    centreAnswersData.Answer5,
+                    centreAnswersData.Answer6,
+                    detailsLastChecked
                 );
 
-                var oldDelegateDetails =
-                    verifiedDelegateUsers.SingleOrDefault(u => u.Id == myAccountDetailsData.DelegateId);
-
-                if (oldDelegateDetails != null && centreAnswersData != null)
-                {
-                    userDataService.UpdateDelegateUserCentrePrompts(
-                        myAccountDetailsData.DelegateId!.Value,
-                        centreAnswersData.Answer1,
-                        centreAnswersData.Answer2,
-                        centreAnswersData.Answer3,
-                        centreAnswersData.Answer4,
-                        centreAnswersData.Answer5,
-                        centreAnswersData.Answer6
-                    );
-
-                    groupsService.SynchroniseUserChangesWithGroups(
-                        oldDelegateDetails,
-                        myAccountDetailsData,
-                        centreAnswersData
-                    );
-                }
+                groupsService.SynchroniseUserChangesWithGroups(
+                    oldDelegateUserDetails!,
+                    myAccountDetailsData,
+                    centreAnswersData

```

Comment:
Here call constructor on ~CentreAnswersData~ RegistrationFieldAnswers to combine `delegateDetailsData` and `jobGroup`